src/miniscript/satisfy.rs

```diff
@@ -109,59 +106,46 @@ pub trait Satisfier<Pk: MiniscriptKey + ToPublicKey> {
     }
 
     /// Assert whether an relative locktime is satisfied
-    fn check_older(&self, _: u32) -> bool {
+    fn check_older(&self, _: Sequence) -> bool {
         false
     }
 
     /// Assert whether a absolute locktime is satisfied
-    fn check_after(&self, _: u32) -> bool {
+    fn check_after(&self, _: LockTime) -> bool {
         false
     }
 }
 
 // Allow use of `()` as a "no conditions available" satisfier
 impl<Pk: MiniscriptKey + ToPublicKey> Satisfier<Pk> for () {}
 
-/// Newtype around `u32` which implements `Satisfier` using `n` as an
-/// relative locktime
-pub struct Older(pub u32);
-
-impl<Pk: MiniscriptKey + ToPublicKey> Satisfier<Pk> for Older {
-    fn check_older(&self, n: u32) -> bool {
-        if self.0 & SEQUENCE_LOCKTIME_DISABLE_FLAG != 0 {
-            return true;
+impl<Pk: MiniscriptKey + ToPublicKey> Satisfier<Pk> for Sequence {
+    fn check_older(&self, n: Sequence) -> bool {
+        if !self.is_relative_lock_time() {
+            return false;
         }
 
-        /* If nSequence encodes a relative lock-time, this mask is
-         * applied to extract that lock-time from the sequence field. */
-        const SEQUENCE_LOCKTIME_MASK: u32 = 0x0000ffff;
-
-        let mask = SEQUENCE_LOCKTIME_MASK | SEQUENCE_LOCKTIME_TYPE_FLAG;
-        let masked_n = n & mask;
-        let masked_seq = self.0 & mask;
-        if masked_n < SEQUENCE_LOCKTIME_TYPE_FLAG && masked_seq >= SEQUENCE_LOCKTIME_TYPE_FLAG {
-            false
+        if self.is_height_locked() && n.is_height_locked()
+            || self.is_time_locked() && n.is_time_locked()
+        {
+            n.to_consensus_u32() <= self.to_consensus_u32()
```

Comment:
The previous implementation masked the sequence before doing the comparison. How can we justify not doing this here? What if one has the RBF disabled  and the other doesn't for example (disabling RBF will always be the highest value unless you mask out that bit).

PS methods on `Sequence` like `is_satisfied_by_height` and `is_satisfied_by_time` would be super useful here.

Reply:
> How can we justify not doing this here?

Thanks man, you are 100% correct, this code is not the same as the code it replaces - that's my bug.

> PS methods on Sequence like is_satisfied_by_height and is_satisfied_by_time would be super useful here.

A relative lock time type is meant to be in the works (not sure if anyone is actively working on it yet) and that type would have these methods you suggest. When `Sequence` was added relative lock time methods were explicitly out of scope.

Reply:
I've reverted these changes, so we now have
```rust
        // We need a relative lock time type in rust-bitcoin to clean this up.

        /* If nSequence encodes a relative lock-time, this mask is
         * applied to extract that lock-time from the sequence field. */
        const SEQUENCE_LOCKTIME_MASK: u32 = 0x0000ffff;
        const SEQUENCE_LOCKTIME_TYPE_FLAG: u32 = 0x00400000;

        let mask = SEQUENCE_LOCKTIME_MASK | SEQUENCE_LOCKTIME_TYPE_FLAG;
        let masked_n = n.to_consensus_u32() & mask;
        let masked_seq = self.to_consensus_u32() & mask;
        if masked_n < SEQUENCE_LOCKTIME_TYPE_FLAG && masked_seq >= SEQUENCE_LOCKTIME_TYPE_FLAG {
            false
        } else {
            masked_n <= masked_seq
        }
```

Reply:
Please see https://github.com/rust-bitcoin/rust-miniscript/pull/455 which implements this using a new `relative::LockTime` API.